src/rules/creates-lead-salesforce.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @overview On first login call the Salesforce API to record the contact as a new Lead.
+ * @gallery true
+ * @category webhook
+ * 
+ * Creates a new Lead in Salesforce on First Login
+ * 
+ * This rule will check if this is the first user login, and in that case will call Salesforce API to record the contact as a new Lead. It is using Salesforce REST APIs and the `resource owner` flow to obtain an `access_token`.
+ * The username you use to authenticate the API will appear as the __creator__ of the lead.
+ * 
+ * Note: this sample implements very basic error handling.
+ * 
+ */
+
+function (user, context, done) {
+  user.app_metadata = user.app_metadata || {};
+  if (user.app_metadata.recordedAsLead) {
+    return done(null,user,context);
+  }
+
+  const MY_SLACK_WEBHOOK_URL = 'YOUR SLACK WEBHOOK URL';
+  const slack = require('slack-notify')(MY_SLACK_WEBHOOK_URL);
+
+  //Populate the variables below with appropriate values
+  const SFCOM_CLIENT_ID = configuration.SALESFORCE_CLIENT_ID;
+  const SFCOM_CLIENT_SECRET = configuration.SALESFORCE_CLIENT_SECRET;
+  const USERNAME = configuration.SALESFORCE_USERNAME;
+  const PASSWORD = configuration.SALESFORCE_PASSWORD;
+  getAccessToken(
+    SFCOM_CLIENT_ID,
+    SFCOM_CLIENT_SECRET,
+    USERNAME,
+    PASSWORD,
+    (r) => {
```

Comment:
can we change these `r` (and `e` below) references to be something more descriptive?